sdk/core/azure-core/inc/http/curl/curl.hpp

```diff
@@ -19,6 +21,9 @@ namespace Azure { namespace Core { namespace Http {
     // This can be customizable in the HttpRequest
     constexpr int64_t c_UploadDefaultChunkSize = 1024 * 64;
     constexpr auto c_LibcurlReaderSize = 1024;
+    // A connection will expire if it is not used during 90 sec. Server would close connection
+    // already.
+    constexpr double c_MaxSecondsForKeepAliveConnection = 90;
```

Comment:
I think server is at liberty to decide when to close the connection. We cannot make the assumption that the connection isn't closed within 90 secs and will be closed after 90 seconds. We have to come up with some way to probe.

Reply:
we can set it to something lower.

Some alternative is to always try to re-use the connection and react to a closed connection using the retry policy to spawn a new connection and free the one that fail.  

I will see if Server is returning a header with information about session timeout like:
https://stackoverflow.com/questions/12095533/proper-use-of-keepalive-in-apache-htaccess

Reply:
> Some alternative is to always try to re-use the connection and react to a closed connection using the retry policy to spawn a new connection and free the one that fail.

I don't think to rely on the retry policy here is a good idea. We waste one retry count. What if the customer set the retry count to zero to completely disable retry mechanism? 

I think always trying to reuse the connection and reconnecting on failure is the right thing to do here.

> I will see if Server is returning a header with information about session timeout like:
> https://stackoverflow.com/questions/12095533/proper-use-of-keepalive-in-apache-htaccess

But the connection may still be disconnected by server or get interrupted due to network failure before that timeout.

Reply:
This setting was removed.
Connection is open as soon as it gets created.
And if the request fails when trying to send a request due to connection closed or socket lost, a new connection is requested